HeterogeneousCore/CUDACore/interface/ConvertingESProducerT.h

```diff
@@ -11,33 +9,31 @@
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 #include "FWCore/Utilities/interface/typelookup.h"
 
-template <typename Target, typename Source, typename Record>
-class HcalRawESProducerGPU : public edm::ESProducer {
+template <typename Record, typename Target, typename Source>
+class ConvertingESProducerT : public edm::ESProducer {
 public:
-  explicit HcalRawESProducerGPU(edm::ParameterSet const& ps) {
+  explicit ConvertingESProducerT(edm::ParameterSet const& ps) {
     auto const& label = ps.getParameter<std::string>("label");
     auto const& name = ps.getParameter<std::string>("ComponentName");
     auto cc = setWhatProduced(this, name);
     token_ = cc.consumes(edm::ESInputTag{"", label});
   }
 
   std::unique_ptr<Target> produce(Record const& record) {
-    // retrieve conditions in old format
-    auto sourceProduct = record.getTransientHandle(token_);
-
-    return std::make_unique<Target>(*sourceProduct);
+    // retrieve conditions in the old format and build a product in the new format
+    return std::make_unique<Target>(* record.getHandle(token_).product());
```

Comment:
This could be simplified to
```suggestion
    return std::make_unique<Target>(record.get(token_));
```